front/src/routes/dashboard/edit-dashboard/EditBoxColumns.jsx

```diff
@@ -8,145 +9,250 @@ import EditBox from './EditBox';
 import EmptyColumnDropZone from './EmptyColumnDropZone';
 import BottomDropZone from './BottomDropZone';
 import AutoScrollMobile from '../../../components/drag-and-drop/AutoScrollMobile';
-import style from '../style.css';
+import style from './style.css';
+import stylePrimary from '../style.css';
 import { DASHBOARD_VISIBILITY_LIST } from '../../../../../server/utils/constants';
 
 const DASHBOARD_EDIT_BOX_TYPE = 'DASHBOARD_EDIT_BOX';
 
-const EditBoxColumns = ({ children, ...props }) => (
-  <div class="pb-6">
-    <h3>
-      <Text id="dashboard.editDashboardTitle" />
-    </h3>
-    {props.dashboardAlreadyExistError && (
-      <div class="alert alert-danger">
-        <Text id="newDashboard.dashboardAlreadyExist" />
-      </div>
-    )}{' '}
-    {props.dashboardValidationError && (
-      <div class="alert alert-danger">
-        <Text id="newDashboard.validationError" />
-      </div>
-    )}
-    {props.unknownError && (
-      <div class="alert alert-danger">
-        <Text id="newDashboard.unknownError" />
-      </div>
-    )}
-    <div class="row align-items-end">
-      <div class="col-md-4">
-        <div class="form-group">
-          <label class="form-label">
-            <Text id="dashboard.editDashboardNameLabel" />
-          </label>
-          <Localizer>
-            <input
-              type="text"
-              class="form-control"
-              placeholder={<Text id="dashboard.editDashboardNameLabel" />}
-              value={props.homeDashboard.name}
-              onInput={props.updateCurrentDashboardName}
-            />
-          </Localizer>
+class EditBoxColumns extends Component {
+  constructor(props) {
+    super(props);
+    this.columnCardRef = createRef();
+    this.state = {
+      parentHeight: 0,
+      maxBoxes: 3
+    };
+  }
+  componentDidMount() {
+    this.updateHeight();
+    this.mutationObserver = new MutationObserver(() => {
+      this.updateHeight();
+    });
+    this.mutationObserver.observe(this.columnCardRef.current, { childList: true, subtree: true });
+
+    this.resizeObserver = new ResizeObserver(() => {
+      this.updateHeight();
+    });
+    this.resizeObserver.observe(this.columnCardRef.current);
+  }
+
+  componentWillUnmount() {
+    if (this.columnCardRef.current) {
+      this.mutationObserver.disconnect();
+      this.resizeObserver.unobserve(this.columnCardRef.current);
+    }
```

Comment:
Did you write all this code yourself or is it generated by ChatGPT ?

To me all this code should just be a bit of CSS... 

It's completely overkill, Javascript should be used only for business logic, not design! 😅

Reply:
Actually yes and no. I found 'ResizeObserver' on the MDN documentation because I could not achieve the total recovery of the height when updating the boxes (addition/deletion) using css alone.
But maybe you can help me...

For the 'mutationObserver' part, taking into account dashboard view changes did not work. I shared to chatGPT for a quick response.